config/profile.go

```diff
@@ -651,15 +652,19 @@ func (p *Profile) SetRootPath(rootPath string) {
 func (p *Profile) resolveSourcePath(sourceBase string, sourcePaths ...string) []string {
 	var applySourceBase, applyBaseDir pathFix
 
-	// Backup source is NOT relative to the configuration, but to PWD or sourceBase (if not empty)
-	// Applying "sourceBase" if set
-	if sourceBase = strings.TrimSpace(sourceBase); sourceBase != "" {
-		sourceBase = fixPath(sourceBase, expandEnv, expandUserHome)
-		applySourceBase = absolutePrefix(sourceBase)
-	}
-	// Applying a custom PWD eagerly so that own commands (e.g. "show") display correct paths
-	if p.BaseDir != "" {
-		applyBaseDir = absolutePrefix(p.BaseDir)
+	sourceBase = fixPath(sourceBase, expandEnv, expandUserHome)
+	// When "source-relative" is set, the source paths are relative to the "source-base"
+	if p.Backup == nil || !p.Backup.SourceRelative {
+		// Backup source is NOT relative to the configuration, but to PWD or sourceBase (if not empty)
+		// Applying "sourceBase" if set
+		if sourceBase = strings.TrimSpace(sourceBase); sourceBase != "" {
+			applySourceBase = absolutePrefix(sourceBase)
+		}
+		// Applying a custom PWD eagerly so that own commands (e.g. "show") display correct paths
+		if p.BaseDir != "" {
+			applyBaseDir = absolutePrefix(p.BaseDir)
+		}
+
 	}
```

Comment:
At this step we should add the following: 

```go
} else if p.BaseDir == "" && sourceBase == "" && p.config != nil {
    p.config.reportChangedPath(".", "", "source-base (for relative source)")
}
```

This prints a warning when nether `source-base` nor `base-dir` is defined but paths are set to relative.